### PR TITLE
Adapt network preloading strategy when applying modifications

### PR DESCRIPTION
**Pull request by FranckLecuyer**

### CODE REVIEW

**Review by SlimaneAmar — CHANGES_REQUESTED** [review]

File: src/main/java/org/gridsuite/modification/server/service/NetworkModificationService.java [hunk 1]
```diff
@@ -173,12 +183,26 @@ public void deleteNetworkModifications(UUID groupUuid, List<UUID> modificationsU
         }
     }
 
+    private PreloadingStrategy computePreloadingStrategy(List<ModificationInfos> modificationInfos) {
```
[the remark this answers is not in the document]
Comment (FranckLecuyer) [reply src/main/java/org/gridsuite/modification/server/service/NetworkModificationService.java:186]: Done

File: src/main/java/org/gridsuite/modification/server/ModificationType.java [hunk 2]
```diff
@@ -6,35 +6,52 @@
...
+
+    ModificationType(PreloadingStrategy strategy) {
+        this.strategy = strategy;
+    }
+
+    public PreloadingStrategy getStrategy() {
+        return strategy;
+    }
+
+    public ModificationType maxStrategy(ModificationType other) {
```
Comment (SlimaneAmar) [inline src/main/java/org/gridsuite/modification/server/ModificationType.java:53]: Return `PreloadingStrategy`

File: src/main/java/org/gridsuite/modification/server/ModificationType.java [hunk 3]
```diff
@@ -6,35 +6,52 @@
...
+        return ((strategy == PreloadingStrategy.NONE && (other.strategy == PreloadingStrategy.COLLECTION || other.strategy == PreloadingStrategy.ALL_COLLECTIONS_NEEDED_FOR_BUS_VIEW))
+                || (strategy == PreloadingStrategy.COLLECTION && other.strategy == PreloadingStrategy.ALL_COLLECTIONS_NEEDED_FOR_BUS_VIEW)) ? other : this;
```
Comment (SlimaneAmar) [inline src/main/java/org/gridsuite/modification/server/ModificationType.java:55]: Invert the  last test ?

File: src/main/java/org/gridsuite/modification/server/ModificationType.java [hunk 4]
```diff
@@ -6,35 +6,52 @@
...
+    }
```
Comment (SlimaneAmar) [inline src/main/java/org/gridsuite/modification/server/ModificationType.java:56]:
```
    public PreloadingStrategy maxStrategy(ModificationType other) {
        switch (strategy) {
            case NONE:
                if (other.strategy != PreloadingStrategy.NONE) {
                    return other.getStrategy();
                }
            case ALL_COLLECTIONS_NEEDED_FOR_BUS_VIEW:
                if (other.strategy == PreloadingStrategy.COLLECTION) {
                    return other.getStrategy();
                }
            default:
                return strategy;
        }
    }
```

**Pull request merged**